src/components/ContinuationIn/UploadAffidavit.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div>
+    <!-- Upload Affidavit -->
+    <p>
+      Upload the affidavit from the directors.
+    </p>
+
+    <ul>
+      <li>Use a white background and a legible font with contrasting font colour</li>
+      <li>PDF file type (maximum 30 MB file size)</li>
+    </ul>
+
+    <FileUploadPreview
+      class="mt-4"
+      inputFileLabel="Affidavit from directors"
+      :maxSize="MAX_FILE_SIZE"
+      :pdfPageSize="PdfPageSize.LETTER_SIZE"
+      :hint="business.affidavitFile ? 'File uploaded' : undefined"
+      :inputFile="business.affidavitFile"
+      :showErrors="getShowErrors"
+      :customErrorMessage.sync="customErrorMessage"
+      :isRequired="getShowErrors"
+      @fileValidity="onFileValidity($event)"
+      @fileSelected="onFileSelected($event)"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import { Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { StatusCodes } from 'http-status-codes'
+import { DateMixin, DocumentMixin } from '@/mixins'
+import { ExistingBusinessInfoIF, PresignedUrlIF } from '@/interfaces'
+import { PdfPageSize } from '@/enums'
+import FileUploadPreview from '@/components/common/FileUploadPreview.vue'
+@Component({
+  components: {
+    FileUploadPreview
+  }
+})
+export default class UploadAffidavit extends Mixins(DateMixin, DocumentMixin) {
+  readonly PdfPageSize = PdfPageSize
+
+  @Getter(useStore) getExistingBusinessInfo!: ExistingBusinessInfoIF
+  @Getter(useStore) getKeycloakGuid!: string
+  @Getter(useStore) getShowErrors!: boolean
+
+  // The business is the same object that the parent is working with
+  @Prop({ default: null }) readonly business!: ExistingBusinessInfoIF
+
+  // Local properties
+  customErrorMessage = ''
+  fileValidity = false
+
+  /**
+   * Called when FileUploadPreview tells us whether a file is valid.
+   * This is called right before the File Selected event.
+   */
+  onFileValidity (valid: boolean): void {
+    this.fileValidity = valid
+  }
+
+  /**
+   * Called when FileUploadPreview tells us about a new or cleared file.
+   * This is called right after the File Validity event.
+   */
+  async onFileSelected (file: File): Promise<void> {
+    if (file) {
+      if (this.fileValidity) {
+        // try to upload to Minio
+        let psu: PresignedUrlIF
+        try {
+          psu = await this.getPresignedUrl(file.name)
+          const res = await this.uploadToUrl(psu.preSignedUrl, file, psu.key, this.getKeycloakGuid)
+          if (!res || res.status !== StatusCodes.OK) throw new Error()
+        } catch {
+          // put file uploader into manual error mode by setting custom error message
+          this.customErrorMessage = this.UPLOAD_FAILED_MESSAGE
+          this.$forceUpdate() // force file upload component to react
+          return // don't add to list
+        }
+
+        // add properties reactively to business object and emit to parent coponent
```

Comment:
Update comment, please.

The comment on line 50 looks good 👍

Reply:
Updated